File: engine/engine-model/src/main/scala/tools/packet/pet/PetNameChange.scala
package tools.packet.pet

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class PetNameChange(private var _characterId: Int, private var _newName: String, private var _slot: Int) extends PacketInput {
  def characterId: Int = _characterId

  def newName: String = _newName

  def slot: Int = _slot

  override def opcode(): SendOpcode = SendOpcode.PET_NAME_CHANGE
}

File: engine/engine-model/src/main/java/tools/packet/pet/PetNameChange.java
package tools.packet.pet;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record PetNameChange(Integer characterId, String newName, Integer slot) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.PET_NAME_CHANGE;
   }
}